tests/common/mod.rs

```diff
@@ -104,7 +124,188 @@ pub async fn setup_bitcoind() -> BitcoindNode {
+            cert_path: cert_path,
+            macaroon_path: macaroon_path,
+            _handle: cmd,
+            client: None,
+        }
+    }
+
+    // Setup the client we need to interact with the LND node.
+    async fn setup_client(&mut self) {
+        // We need to give lnd some time to start up before we'll be able to interact with it via the client.
+        let mut retry = false;
+        let mut retry_num = 0;
+        while retry_num == 0 || retry {
+            thread::sleep(Duration::from_secs(2));
+
+            let client_result = tonic_lnd::connect(
+                self.address.clone(),
+                self.cert_path.clone(),
+                self.macaroon_path.clone(),
+            )
+            .await;
+
+            match client_result {
+                Ok(client) => {
+                    self.client = Some(client);
+
+                    retry = false;
+                    retry_num += 1;
+                }
+                Err(err) => {
+                    println!(
+                        "getting client error {err}, retrying call {} time",
+                        retry_num
+                    );
+                    if retry_num == 3 {
+                        panic!("could not set up client: {err}")
+                    }
+                    retry = true;
+                    retry_num += 1;
+                }
+            }
+        }
+    }
+
+    #[allow(dead_code)]
+    pub async fn get_info(&mut self) -> tonic_lnd::lnrpc::GetInfoResponse {
+        let resp = if let Some(client) = self.client.clone() {
+            let get_info_req = GetInfoRequest {};
+            let make_request = || async {
+                client
+                    .clone()
+                    .lightning()
+                    .get_info(get_info_req.clone())
+                    .await
+            };
+            let resp = test_utils::retry_async(make_request, String::from("get_info"));
+            resp.await.unwrap()
+        } else {
+            panic!("No client")
+        };
+
+        resp
+    }
+
+    // connect_to_peer connects to the specified peer.
+    pub async fn connect_to_peer(
+        &mut self,
+        node_id: PublicKey,
+        addr: SocketAddr,
+    ) -> tonic_lnd::lnrpc::ConnectPeerResponse {
+        let ln_addr = tonic_lnd::lnrpc::LightningAddress {
+            pubkey: node_id.to_string(),
+            host: addr.to_string(),
+        };
+
+        let connect_req = tonic_lnd::lnrpc::ConnectPeerRequest {
+            addr: Some(ln_addr),
+            timeout: 20,
+            ..Default::default()
+        };
+
+        let resp = if let Some(client) = self.client.clone() {
+            let make_request = || async {
+                client
+                    .clone()
+                    .lightning()
+                    .connect_peer(connect_req.clone())
+                    .await
+            };
+            let resp = test_utils::retry_async(make_request, String::from("connect_peer"));
```

Comment:
I think that we can simplify this code quite a bit if we just LND's state service to make sure the server is up. 

Happy for it to happen in a follow up because it needs a `tonic_lnd` update, but I think that we _do_ actually want to fail if API calls don't succeed once the node is up.

Reply:
Sounds good. Yeah using the SubscribeState API call looks like it'll work. I'll def get that into a follow up PR and put in a PR to add the state subservice to tonic_lnd 👍🏻